### PR TITLE
Fixed Bugs when image is  pasted from clipboard.

### LINKED ISSUE #8880: 2.15beta4 crash on image paste
Submitter entered crash comment "can't paste image into note"

```
java.lang.IllegalArgumentException: Prefix string too short
at java.io.File.createTempFile(File.java:1991)
at java.io.File.createTempFile(File.java:2064)
at com.ichi2.anki.NoteEditor.loadImageIntoCollection(NoteEditor.java:4)
at com.ichi2.anki.NoteEditor.onImagePaste(NoteEditor.java:2)
at com.ichi2.anki.NoteEditor.c(Unknown Source:0)
at com.ichi2.anki.k5.onImagePaste(Unknown Source:2)
at com.ichi2.anki.FieldEditText.onImagePaste(FieldEditText.java:1)
at com.ichi2.anki.FieldEditText.lambda$onCreateInputConnection$0(FieldEditText.java:6)
at com.ichi2.anki.FieldEditText.a(Unknown Source:0)
at com.ichi2.anki.o3.onCommitContent(Unknown Source:9)
at androidx.core.view.inputmethod.InputConnectionCompat$1.commitContent(InputConnectionCompat.java:1)
at com.android.internal.view.IInputConnectionWrapper.executeMessage(IInputConnectionWrapper.java:567)
at com.android.internal.view.IInputConnectionWrapper$MyHandler.handleMessage(IInputConnectionWrapper.java:93)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:246)
at android.app.ActivityThread.main(ActivityThread.java:8506)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
```

### PR DESCRIPTION
**Pull request by prateek-singh-3212**
## Purpose / Description
Handeled IllegalArgumentException occurs while pasting the image from clipboard

## Fixes
Fixes #8880

## Approach
check the length of the file name if the length is less than 3 then I changed the file name.

## How Has This Been Tested?

Tested manually by pasting different images from clipboard.

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 1]
```diff
@@ -1592,13 +1592,20 @@ private boolean onImagePaste(EditText editText, Uri uri) {
...
         final int oneMegabyte = 1 * 1000 * 1000;
         String filename = ContentResolverUtil.getFileName(getContentResolver(), uri);
         InputStream fd = getContentResolver().openInputStream(uri);
 
         Map.Entry<String, String> fileNameAndExtension = FileUtil.getFileNameAndExtension(filename);
 
-        File clipCopy = File.createTempFile(fileNameAndExtension.getKey(), fileNameAndExtension.getValue());
+        if (checkFilename(fileNameAndExtension)) {
+            fileName = String.format("%s-name", fileNameAndExtension.getKey());
+        }else {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1604]: spacing nit, trivial

```suggestion
        } else {
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
Definitely fixes the prefix one, we just weren't following the API contract:

https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/io/File.html#createTempFile(java.lang.String,java.lang.String)

> must be at least three characters long

Oops :sweat_smile: . Great fix

I agree we are already handling the other exception, but the way it is being reported will continue to trigger false alarms for me and I'll waste my time and maybe others if it stays the same.

This line https://github.com/ankidroid/Anki-Android/blob/07df23073e7a1645700adf5483dce1c7651aada4/AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java#L1579

should look more like this one https://github.com/ankidroid/Anki-Android/pull/8844/files#diff-f5fe2d7c59f0efdb3d639d6032040921173f1f57e7b3ae9be278d3d277218280R2189

And then we'd have all the diagnostics we need (assuming we include them in the message) but it will be labeled "ManuallyReportedException" so it won't trigger high priority effort (unless we get a lot of them...) while we figure out what to do in those non-crash cases. It's just for future triage once we have data

Could you make a change like that? Then this is good to go

**Comment by prateek-singh-3212** [comment]
Yup! I will make changes as stated above.

**Review by mikehardy — APPROVED** [review]
Nice! This should be easy to track down now  - monitoring frequency etc without causing too many false alarms. Thanks!

**Pull request merged**